### PR TITLE
[5.2] Password Strength Symbols

### PR DESCRIPTION
**Pull request by brianteeman**
The script used to check for symbols in a password contains a limited set of symbols.

Pull Request for Issue #41552 .

### Summary of Changes
This PR updates the script to ensure that all of the the list of special characters/symbols provided by [OWASP](https://owasp.org/www-community/password-special-characters) are also included. They are the punctuation characters that are present on standard US keyboard.
The entire set of special characters/symbols is now @$!#?=;:*-_€%&()`´+[]{}'"|,.<>/~^ 

I added the additional comment to the licence so that it is clear that the script has been changed from the original which probably  should have been done earlier when the regex was updated to include @

### Testing Instructions
This change will require using a prebuilt package or `npm ci` to test

Set the password requirements in the User Options Password Minimum Symbols to 1

Try to create a password containing letters and one or more symbols from this list 
+[]{}'"|,.<>\/~^

### Actual result BEFORE applying this Pull Request
The password will be rejected

### Expected result AFTER applying this Pull Request
The password will be accepted

### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 388d8e3709dd4c66c23a0e23a4719950fba1561c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43484">issues.joomla.org/tracker/joomla-cms/43484</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 388d8e3709dd4c66c23a0e23a4719950fba1561c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43484">issues.joomla.org/tracker/joomla-cms/43484</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43484">issues.joomla.org/tracker/joomla-cms/43484</a>.</sub>

**Comment by Quy** [comment]
Please fix javascript cs https://ci.joomla.org/joomla/joomla-cms/76291/1/20

File: build/media_source/system/js/fields/passwordstrength.es6.js [hunk 1]
```diff
@@ -50,7 +52,7 @@ class PasswordStrength {
     // eslint-disable-next-line no-useless-escape
     score += this.constructor.calc(
       value,
-      /[@$!#?=;:*\-_€%&()`´]/g,
+      /[@$!#?=;:*\-_€%&()`´+\[\]{}'"\\|,.<>\/~^]/g,
```
Comment (Quy) [inline build/media_source/system/js/fields/passwordstrength.es6.js]:
```suggestion
      /[@$!#?=;:*\-_€%&()`´+[\]{}'"\\|,.<>/~^]/g,
```

**Comment by pe7er** [comment]
Thank you @brianteeman and @Quy !

**Comment by MaverickPOZ** [comment]
The symbol "_" is not correctly recognized as a sybmol, so password "0123456789_Aa" is incorrect and throw error that mininum 1 special character (symbol) is required (only after registration form submit). Progress bar is green while paste password. Can You fix that ?

**Comment by brianteeman** [comment]
@MaverickPOZ please do not comment on closed PR - hardly anyone will see it. Please create a new issue

**Comment by petrovicivan** [comment]
Default installation shows 0 in Password Options->Minimum Symbols

**Comment by MaverickPOZ** [comment]
And so what ? This is default ... Please put 1 in that field and use _ in password.

**Comment by petrovicivan** [comment]
https://docs.joomla.org/J5.x:Enhancing_Password_Security_with_Symbolic_Characters#Additional_Notes

> This change applies to all Joomla 5.2 installations. Administrators should ensure that the password requirements are updated in the configuration to take full advantage of the enhanced security features.

I would rather accept what is written in the documentation

**Comment by MaverickPOZ** [comment]
Ivan, I have fresh J5.2 and nothing change - please provide password with symbol _ "TestPassword123With_" at https://demo.webster-studio.pl/login?view=registration - the bar is green but after submit form the error with "no symbol" appear.

**Comment by MaverickPOZ** [comment]
Screenshot
Screenshot 2024-10-21 at 09-21-55 Login - Joomla 5 Demo Webster-Studio

**Comment by ceford** [comment]
This is a separate issue. The client side password has passed validation there but the server side uses this:

`            $nsymbols = preg_match_all('[\W]', $value, $smatch);`

on line 133 of libraries/src/Form/Rule/PasswordRule.php - I think underline does not count as a non-word character.

**Pull request merged**